### PR TITLE
make app movable to sd card

### PR DESCRIPTION
**Pull request by TacoTheDank**

### CODE REVIEW

**Comment by PiwwowPants** [comment]
After external storage is unmounted and remounted, my impression is Services won't restart and most notably AlarmManager alarms are removed. 

AlarmManager seems used for the AutoCache task. After remounting and opening the app again, would this AlarmManager alarm get set again or would it require the user set AutoCache up again (toggle off and on)?

Or are any of my assumptions about external storage here wrong. @ccrama

[Reference](https://developer.android.com/guide/topics/data/install-location). Old so it may have changed.

**Comment by PiwwowPants** [comment]
I'm not very sure how to respond to that BUT my main point of bringing this up is that phones are moving more and more away from external storage, with some reasons being good. Because of that it doesn't seem Google would or does put priority on it. 

I'd hate this to go in and issue counts go up because of it since a user may not even be aware it's on external storage so asking that question may complicate any sort of resolution. Additionally debugging would be difficult for those without external storage phones

If you or someone out there with a phone with external storage could verify that remounting external storage during up-time of the phone doesn't cripple the services the app runs, with explicit attention given to the auto-caching feature, it'd be much more comfortable to accept as a change, but I think more analysis may need done to see how the app responds to that in order to prevent user issues increases because of this change.

**Comment by TacoTheDank** [comment]
the reason i even made this issue in the first place is because i never seem to have enough storage and also i have a 128GB micro sd card lol so sure ill try to remember to test it

**Comment by TacoTheDank** [comment]
ill restate what i need to do in case i have it wrong:
Build Slide apk from my fork and install onto phone, and move it to sd card
Go into Slide and set up AlarmManager and verify that it's working by having an alarm go off
Check apps settings -> running processes just to make sure it's running
Close Slide
Unmount and remount sd card
Check apps settings -> running processes to see if it started up again

**Comment by PiwwowPants** [comment]
If you can verify that the AlarmManager for auto-caching Subreddits is running still from App Settings on your device, yeah. 

Assuming that isn't possible though, you'd need to actually wait for the AlarmManager to fire naturally (you can set the time for caching or modify when it'll fire to make your life better).

**Comment by PiwwowPants** [comment]
I forgot that I can't just not submit my code reviews. I made these a few days ago and just now realized that I didn't submit them. Sorry about that.

**Comment by ccrama** [comment]
Slide already allows for this, https://github.com/ccrama/Slide/blob/23b5f51f07b2de51ccdffa2f2b3d6301d862ae96/app/src/main/AndroidManifest.xml#L18

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest package="me.ccrama.redditslide"
-          xmlns:android="http://schemas.android.com/apk/res/android">
-
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="me.ccrama.redditslide" android:installLocation="auto">
+          
+package="taco.apkmirror" android:installLocation="auto">
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/src/main/AndroidManifest.xml]: lmao i forgot to delete that

**Comment by TacoTheDank** [comment]
my phone says otherwise screenshot_2018-07-19-19-01-55

**Comment by TacoTheDank** [comment]
and im sure its because the install location has to be specified next to the package name

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest package="me.ccrama.redditslide"
-          xmlns:android="http://schemas.android.com/apk/res/android">
-
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="me.ccrama.redditslide" android:installLocation="auto">
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/src/main/AndroidManifest.xml]: not sure what you mean by that, example probably necessary

**Comment by TacoTheDank** [comment]
at any rate ill make a few updates and generate an apk once everything is ok, and ill upload it here

**Comment by PiwwowPants** [comment]
I believe @TacoTheDank is correct  That parameter should be on the manifest element, not application, per Android's spec at least.

**Comment by TacoTheDank** [comment]
in case you were wondering, my phone's a Samsung Galaxy S4 (AT&T carrier model) that runs 5.0.1

**Comment by TacoTheDank** [comment]
oh did i say a few updates woops i meant impulsively update a *whole bunch* of stuff i didnt need to

why am i like this

regardless, here's the link to the apks i built from that commit: https://app.box.com/s/41mfzn697ld4uiltzuv23ccz2f10oksq

**Comment by TacoTheDank** [comment]
as far as i can tell, everything works fine. besides, people dont often unmount and remount their sd cards so thats not really an issue to be concerned about anyways

**Comment by TacoTheDank** [comment]
I completely forgot about this, and oh god it's hideous. I'll just close this xd

**Pull request closed without merging**